core/runtime/src/syscalls_impl_riscv.rs

```diff
@@ -0,0 +1,82 @@
+use libtock_platform::{OneArgMemop, RawSyscalls, YieldType, ZeroArgMemop};
+
+impl RawSyscalls for crate::TockSyscalls {
+    // This yield implementation is currently limited RISC-V versions without
+    // floating-point registers, as it does not mark them clobbered.
+    #[cfg(not(any(target_feature = "d", target_feature = "f")))]
+    fn raw_yield(r0_in: YieldType) -> u32 {
```

Comment:
Shouldn't this return a usize?

Reply:
I went back and forth on some of these types while drafting `RawSyscalls`. `RawSyscalls` is designed for 32-bit systems, as Tock does not have a 64-bit ABI. In that context, `u32` and `usize` are the same.

`u32` and `usize` are different in the unit test environment, as that environment typically runs on an `x86_64` system. In that case, returning a `u32` is a more accurate representation of the values that `TockSyscalls` can return. In other words: I don`t want `libtock_platform` to have to handle the case where a `RawSyscalls` implementations returns a value that is greater than `u32::MAX`.

Reply:
There is on going work to support 64-bit though, so it would be nice if this would work with that.

Reply:
Is there work on a 64-bit syscall ABI yet? Whether this should be `u32` or `usize` depends on what that syscall ABI looks like (e.g. is the return variant still communicated via its own register?)

Reply:
There isn't an ABI yet, but based on the other patches adding 64-bit support to Tock I suspect we would follow the same ABI

Reply:
I feel that discussion is outside the scope of this PR. I'm happy to have that discussion, but as the ABI isn't specified yet I believe the Tock 2.0 TRD the place to have it.

Reply:
Is there a harm in making it a `usize`? I would expect that 64-bit platforms will return a 64-bit value here, so `usize` seems to be more future proof.

Reply:
It makes my unit test library more complex, yes.

Reply:
Given I've asked @bradjc about this on the call, I think it's safe to assume we limit ourselves to 32-bit ABIs for Tock 2.0. After that I have a strong feeling we're going to want to support 64-bit systems (it's also a great way to check that throughout the kernel we're using appropriate integer sizes). I think it's clean to not have these changes interleaved with the rest of the 2.0 changes, especially if it makes testing easier for now.

Reply:
It just feels like we are kicking the can down the road here and are going to hit API breakages as soon as 64-bit support is added